wrapper/src/main/java/software/amazon/jdbc/ds/AwsWrapperDataSource.java

```diff
@@ -94,111 +98,113 @@ public Connection getConnection(final String username, final String password) th
     final Properties props = PropertyUtils.copyProperties(this.targetDataSourceProperties);
     String finalUrl;
 
-    // Identify the URL for connection.
-    if (!StringUtils.isNullOrEmpty(this.jdbcUrl)) {
+    TelemetryFactory telemetryFactory = new DefaultTelemetryFactory(props);
+    TelemetryContext context = telemetryFactory.openTelemetryContext(
+        "software.amazon.jdbc.ds.AwsWrapperDataSource.getConnection",
+        TelemetryTraceLevel.TOP_LEVEL);
 
-      finalUrl = this.jdbcUrl;
-      parsePropertiesFromUrl(this.jdbcUrl, props);
-      setDatabasePropertyFromUrl(props);
-
-      // Override credentials with the ones provided through the data source property.
-      setCredentialProperties(props);
-
-      // Override database with the one provided through the data source property.
-      if (!StringUtils.isNullOrEmpty(this.database)) {
-        PropertyDefinition.DATABASE.set(props, this.database);
-      }
-
-    } else {
-      String serverName = !StringUtils.isNullOrEmpty(this.serverName)
-          ? this.serverName
-          : props.getProperty(SERVER_NAME);
-      String serverPort = !StringUtils.isNullOrEmpty(this.serverPort)
-          ? this.serverPort
-          : props.getProperty(SERVER_PORT);
-      String databaseName = !StringUtils.isNullOrEmpty(this.database)
-          ? this.database
-          : PropertyDefinition.DATABASE.getString(props);
-
-      if (!StringUtils.isNullOrEmpty(serverName)) {
-
-        if (StringUtils.isNullOrEmpty(this.jdbcProtocol)) {
-          throw new SQLException(Messages.get("AwsWrapperDataSource.missingJdbcProtocol"));
-        }
+    try {
 
-        int port = HostSpec.NO_PORT;
-        if (!StringUtils.isNullOrEmpty(serverPort)) {
-          port = Integer.parseInt(serverPort);
-        }
+      // Identify the URL for connection.
+      if (!StringUtils.isNullOrEmpty(this.jdbcUrl)) {
 
-        finalUrl = buildUrl(
-            this.jdbcProtocol,
-            serverName,
-            port,
-            databaseName);
+        finalUrl = this.jdbcUrl;
+        parsePropertiesFromUrl(this.jdbcUrl, props);
+        setDatabasePropertyFromUrl(props);
 
         // Override credentials with the ones provided through the data source property.
         setCredentialProperties(props);
 
         // Override database with the one provided through the data source property.
-        if (!StringUtils.isNullOrEmpty(databaseName)) {
-          PropertyDefinition.DATABASE.set(props, databaseName);
+        if (!StringUtils.isNullOrEmpty(this.database)) {
+          PropertyDefinition.DATABASE.set(props, this.database);
         }
 
       } else {
-        throw new SQLException(Messages.get("AwsWrapperDataSource.missingTarget"));
+        String serverName = !StringUtils.isNullOrEmpty(this.serverName)
```

Comment:
```suggestion
        final String serverName = !StringUtils.isNullOrEmpty(this.serverName)
```
Same for the other variables